src/views/DandisetLandingView/DandisetMain.vue

```diff
@@ -366,6 +366,17 @@ export default {
       delete extra_obj.assetsSummary.schemaKey;
       delete extra_obj.assetsSummary.numberOfBytes;
       delete extra_obj.assetsSummary.numberOfFiles;
+      _.map(extra_obj, (val) => {
+        if (Array.isArray(val) && val.length !== 0) {
+          _.map(val, (each_val) => {
+            if (each_val instanceof Object && Object.keys(each_val).includes('schemaKey')) {
```

Comment:
This can be simplified to:

```suggestion
            if (each_val.hasOwnProperty('schemaKey')) {
```

I am also not sure the condition is even necessary. This first checks to see if the `schemaKey` property exists and removes it if so. I think you can skip straight to the `delete` statement and skip the conditional check.

Reply:
what if the value is not an Object?

Reply:
It's probably not harmful to check whether it's an `Object`, but the check for `schemaKey` isn't needed.

You should keep in mind that other things in JavaScript are also objects, specifically arrays and functions.

Reply:
if I use `each_val.hasOwnProperty('schemaKey')` , I get this ESLint message:
`ESLint: Do not access Object.prototype method 'hasOwnProperty' from target object.(no-prototype-builtins)`.

How does it matter if I use `Object.keys(each_val).includes('schemaKey')` rather than `each_val.hasOwnProperty('schemaKey')` ? The first one doesn't throw a linting error too.

Reply:
It will be important to Google some of these topics as your work in order to build an understanding of best practices. In this case, I searched on the linting message you got and found this [documentation page](https://eslint.org/docs/rules/no-prototype-builtins) that explains the dangers of calling methods like `.hasOwnProperty()` directly, and what to do instead.

To answer your question, `Object.keys(each_val).includes()` has O(n) runtime, and though it likely will not make a difference in this case, it is better to do the standard thing: in this case you solve the performance problem, and you write code that adheres to best practices and thus will cause less burden on reviewers, other developers, and even yourself in the future if and when you come back to this code.

Reply:
thank you for the explanation.